tests/db_functions/math/test_atan.py

```diff
@@ -0,0 +1,44 @@
+import math
+from decimal import Decimal
+
+from django.db.models import DecimalField
+from django.db.models.functions import ATan
+from django.test import TestCase
+
+from ..models import DecimalModel, FloatModel, IntegerModel
+
+
+class ATanTests(TestCase):
+
+    def test_decimal(self):
+        DecimalModel.objects.create(n1=Decimal('-12.9'), n2=Decimal('0.6'))
+        obj = DecimalModel.objects.annotate(n1_atan=ATan('n1'), n2_atan=ATan('n2')).first()
+        self.assertAlmostEqual(obj.n1_atan, math.atan(obj.n1))
+        self.assertAlmostEqual(obj.n2_atan, math.atan(obj.n2))
+
+    def test_float(self):
+        FloatModel.objects.create(f1=-27.5, f2=0.33)
+        obj = FloatModel.objects.annotate(f1_atan=ATan('f1'), f2_atan=ATan('f2')).first()
+        self.assertAlmostEqual(obj.f1_atan, math.atan(obj.f1))
+        self.assertAlmostEqual(obj.f2_atan, math.atan(obj.f2))
+
+    def test_integer(self):
+        IntegerModel.objects.create(small=-20, normal=15, big=-1)
+        obj = IntegerModel.objects.annotate(
+            small_atan=ATan('small'),
+            normal_atan=ATan('normal'),
+            big_atan=ATan('big'),
+        ).first()
+        self.assertAlmostEqual(obj.small_atan, math.atan(obj.small))
+        self.assertAlmostEqual(obj.normal_atan, math.atan(obj.normal))
+        self.assertAlmostEqual(obj.big_atan, math.atan(obj.big))
+
+    def test_transform(self):
+        try:
+            DecimalField.register_lookup(ATan)
+            DecimalModel.objects.create(n1=Decimal('3.12'), n2=Decimal('0'))
+            DecimalModel.objects.create(n1=Decimal('-5'), n2=Decimal('0'))
+            objs = DecimalModel.objects.filter(n1__atan__gt=0)
+            self.assertQuerysetEqual(objs, [float(3.12)], lambda a: float(a.n1))
```

Comment:
The type casting here is weird - `float(3.12)` is pointless as it can be written `3.12`.
I suspect that you shouldn't need `float` in the lambda either as you didn't do it for `ASin`...

Reply:
I have to say it is weird. The reason I wrote float(3.12) is that is I didn't add it, the `[3.12]` gives the result `[Decimal(3.12]` and the lambda function gives result `[3.12]`. I am wondering why `[3.12]` gives the result `[Decimal(3.12]`.

Reply:
I found it will work if I change it to `self.assertQuerysetEqual(objs, [3.12], lambda a: float(a.n1))`. It seems that the type of 3.12 is somehow related to the return type of the lambda function.